src/logging/log-keys-to-redact.test.ts

```diff
@@ -0,0 +1,80 @@
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+
+import { buildKeyToRedact } from './build-key-to-redact';
+import { logKeysToRedact, LogKeysToRedactOptions } from './log-keys-to-redact';
+
+describe('logKeysToRedact', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const options: Omit<LogKeysToRedactOptions, 'redactLogs'> = {
+    clientRequest: {
+      logKey: valueGenerator.string(),
+      headersLogKey: valueGenerator.string(),
+    },
+    outgoingRequest: {
+      logKey: valueGenerator.string(),
+      headersLogKey: valueGenerator.string(),
+    },
+    error: {
+      logKey: valueGenerator.string(),
+      sensitiveChildKeys: [valueGenerator.string(), valueGenerator.string()],
+    },
+    dbError: {
+      logKey: valueGenerator.string(),
+      sensitiveChildKeys: [valueGenerator.string(), valueGenerator.string()],
+    },
+  };
+
+  describe('when redactLogs is false', () => {
+    it('returns an empty array', () => {
+      expect(logKeysToRedact({ redactLogs: false, ...options })).toStrictEqual([]);
+    });
+  });
+
+  describe('when redactLogs is true', () => {
```

Comment:
I think there are more tests we can test here, like testing that the `dbError` keys are included in the keys to redact

Reply:
I added dbError key tests